src/common/components/dateRangePicker/DateRangePicker.tsx

```diff
@@ -0,0 +1,144 @@
+import "react-datepicker/dist/react-datepicker.css";
+import "./datePicker.scss";
+
+import fi from "date-fns/locale/fi";
+import sv from "date-fns/locale/sv";
+import React, { FunctionComponent, MutableRefObject } from "react";
+import DatePicker, { registerLocale } from "react-datepicker";
+import { useTranslation } from "react-i18next";
+
+import { formatDate } from "../../../util/dateUtils";
+import DateRangeInputs from "./DateRangeInputs";
+
+registerLocale("fi", fi);
+registerLocale("sv", sv);
+
+interface Props {
+  endDate: Date | null;
+  locale: string;
+  onChangeEndDate: (date: Date | null) => void;
+  onChangeStartDate: (date: Date | null) => void;
+  startDate: Date | null;
+}
+
+const DateRangePicker: FunctionComponent<Props> = ({
+  endDate,
+  locale,
+  onChangeEndDate,
+  onChangeStartDate,
+  startDate
+}) => {
+  const { t } = useTranslation();
+  // References to input fields to set focus
+  const datePicker = React.useRef(null);
+  const endDateRef = React.useRef<HTMLInputElement>(null);
+  const startDateRef = React.useRef<HTMLInputElement>(null);
+  // Variable to keep track on date field
+  const [counter, setCounter] = React.useState(1);
```

Comment:
a bit more comment maybe
// counter === 1 => start date selected
// counter === 2 => end date selected

Reply:
And just a bit curious how's the different between this 1 2 and start / end basically?